humblegen/src/ast.rs

```diff
@@ -106,26 +122,88 @@ impl VariantDef {
     }
 }
 
-/// A field node (field definition inside struct).
 #[derive(Debug)]
+pub struct ServiceDef {
+    pub name: String,
+    pub doc_comment: Option<String>,
+    pub endpoints: Vec<ServiceEndpoint>,
+}
+
+#[derive(Debug)]
+pub struct ServiceEndpoint {
+    pub doc_comment: Option<String>,
+    pub route: ServiceRoute,
+}
+
+#[derive(Debug)]
+pub enum ServiceRoute {
+    Get {
+        components: Vec<ServiceRouteComponent>,
+        query: Option<TypeIdent>,
+        ret: TypeIdent,
+    },
+    Post {
+        components: Vec<ServiceRouteComponent>,
+        query: Option<TypeIdent>,
+        body: TypeIdent,
+        ret: TypeIdent,
+    },
+}
+
+impl ServiceRoute {
+    pub fn components(&self) -> &Vec<ServiceRouteComponent> {
+        match self {
+            ServiceRoute::Get { components, .. } => components,
+            ServiceRoute::Post { components, .. } => components,
+        }
+    }
+    pub fn query(&self) -> &Option<TypeIdent> {
+        match self {
+            ServiceRoute::Get { query, .. } => query,
+            ServiceRoute::Post { query, .. } => query,
+        }
+    }
+    pub fn rety_type(&self) -> &TypeIdent {
```

Comment:
I believe we can spell out `return_type` here (and in the struct above)? No need to introduce jargon here, I think.